src/back-end/lib/resources/proposal/code-with-us.ts

```diff
@@ -849,15 +849,25 @@ const update: crud.Update<
               session
             );
             break;
-          case "disqualify":
+          case "disqualify": {
             dbResult = await db.updateCWUProposalStatus(
               connection,
               request.params.id,
               CWUProposalStatus.Disqualified,
               body.value,
               session
             );
+            // Check if opportunity should be moved to "Processing" status after disqualification
+            const opportunityId = dbResult.value?.opportunity.id;
+            if (opportunityId) {
+              await db.checkAndUpdateCWUOpportunityProcessingStatus(
```

Comment:
This call and its SWU/TWU equivalents aren't part of a transaction; might be an issue if the disqualification is successful, but checkAndUpdateCWUOpportunityProcessingStatus throws an error.